dashboard/app/helpers/levels_helper.rb

```diff
@@ -220,83 +212,119 @@ def blockly_options(local_assigns={})
       is_project_level
       failure_message_override
     ).map{ |x| x.include?(':') ? x.split(':') : [x,x.camelize(:lower)]}]
-    .each do |dashboard, blockly|
-      # Select first valid value from 1. local_assigns, 2. property of @level object, 3. named instance variable, 4. properties json
+        .each do |dashboard, blockly|
+      # Select value from extra options, or properties json
       # Don't override existing valid (non-nil/empty) values
-      property = local_assigns[dashboard.to_sym].presence ||
-        @level[dashboard].presence ||
-        instance_variable_get("@#{dashboard}").presence ||
```

Comment:
By removing this you'll break anything that sets an instance var to override a blockly option, for example: https://github.com/code-dot-org/code-dot-org/blob/8001ce050a361cee01abacb055cec1fe51bdf1de/dashboard/app/controllers/levels_controller.rb#L186

(Causes e.g. http://localhost:3000/levels/746/embed_level to render incorrectly because `appOptions.hideSource` is null.)

Ultimately this is a great thing to remove :+1: because it will get rid of subtle bugs like this one.  But you'll probably want to do a quick pass over all entries in the big blockly options `%w()` array to make sure no other callers are setting `@instance` variables and expecting them to show up in `appOptions`.